projects/sbb-esta/angular-business/chip/src/chip-input/chip-input.component.ts

```diff
@@ -0,0 +1,238 @@
+      this._value = newValue;
+    }
+  }
+  private _value: any;
+
+  /**
+   * Implemented as part of FormFieldControl.
+   * @docs-private
+   */
+  get inputId() {
+    return this.id;
+  }
+
+  /** Whether the select has a value. */
+  get empty(): boolean {
+    return !this.value || this.value.isEmpty();
+  }
+
+  /**
+   * Event that emits whenever the raw value of the select changes. This is here primarily
+   * to facilitate the two-way binding for the `value` input.
+   * @docs-private
+   */
+  @Output() readonly valueChange: EventEmitter<any> = new EventEmitter<any>();
+
+  constructor(
+    @Self() @Optional() public ngControl: NgControl,
+    private _elementRef: ElementRef,
+    private _changeDetectorRef: ChangeDetectorRef
+  ) {
+    if (this.ngControl) {
+      // Note: we provide the value accessor through here, instead of
+      // the `providers` to avoid running into a circular import.
+      this.ngControl.valueAccessor = this;
+    }
+
+    // Force setter to be called in case id was not specified.
+    this.id = this.id;
+  }
+
+  focused = false;
+  errorState: boolean;
+  inputModel = '';
+  origin = new AutocompleteOriginDirective(this._elementRef);
+
+  private _onTouchedCallback: () => void;
+  private _onChangeCallback: (_: any) => void;
+
+  /** Whether filling out the select is required in the form. */
+  private _required = false;
+
+  /** The aria-describedby attribute on the select for improved a11y. */
+  private _ariaDescribedby: string;
+
+  /** Unique id for this input. */
+  private _uid = `sbb-select-${nextUniqueId++}`;
+
+  /** Emits when the state of the option changes and any parents have to be notified. */
+  readonly stateChanges = new Subject<void>();
+
+  ngOnInit(): void {
+    if (this.autocomplete) {
+      this.autocomplete.optionSelected.subscribe(event => this.onSelect(event.option.value));
+    }
+  }
+
+  /**
+   * Sets the chip input's value. Part of the ControlValueAccessor interface
+   * required to integrate with Angular's core forms API.
+   *
+   * @param value New value to be written to the model.
+   */
+  writeValue(value: string[]): void {
+    if (value) {
+      this.value = value;
+    } else {
+      this.value = [];
+    }
+  }
+
+  /**
+   * Disables the chip input. Part of the ControlValueAccessor interface required
+   * to integrate with Angular's core forms API.
+   *
+   * @param isDisabled Sets whether the component is disabled.
+   */
+  setDisabledState(isDisabled: boolean): void {
+    this.disabled = isDisabled;
+    this._changeDetectorRef.markForCheck();
+    this.stateChanges.next();
+  }
+
+  /**
+   * Adds a given value to the current selected values.
+   */
+  onSelect(option: string) {
+    if (!this.value) {
+      this.value = [];
+    }
+    if (!this.value.includes(option)) {
+      this.writeValue(this.value.concat([option]));
+      this._onChangeCallback(this.value);
+      this._onTouchedCallback();
+    }
+    setTimeout(() => {
```

Comment:
Ok, I'll try to keep that in mind for future implementations. I do try to avoid using `setTimeout` and `Promise.resolve()` as much as possible. Same goes for direct DOM manipulation. I still feel like using `@ViewChild` is the most readable and performant way to implement this little reset. What do you think?

Reply:
Trying to avoid setTimeout, Promise.resolve() and direct DOM manipulation is a good approach. But sometimes it is the most reliable way.
Yes, I think so as well.